### PR TITLE
Making some suggestions about code structure.

### PR DESCRIPTION
**Pull request by jeff-cohere**

### CODE REVIEW

File: swe_roe/ex2.c [hunk 1]
```diff
@@ -8,418 +8,461 @@ static char help[] = "Partial 2D dam break problem.\n";
...
 
-  PetscReal *length; /* length of the edge                        */
+  PetscInt vertices_per_cell  = 4;
+  PetscInt edges_per_cell     = 4;
+  PetscInt neighbors_per_cell = 4;
+
+  PetscCall(RDyAlloc(PetscInt, num_cells, &cells->ids));
+  PetscCall(RDyAlloc(PetscInt, num_cells, &cells->global_ids));
+  PetscCall(RDyAlloc(PetscInt, num_cells, &cells->natural_ids));
+  RDyFill(PetscInt, cells->global_ids, num_cells, -1);
```
Comment (bishtgautam) [inline swe_roe/ex2.c]: It looks a little odd that there isn't a return value from `RDyFill`. How about the following?

```

#define RDyFill(type, array, count, value) \
  for (size_t i = 0; i < (count); ++i) {   \
    *(array+i) = (type)value;                \
  }


RDyFillInt(PetscInt *array, PetscInt count, PetscInt value) {
  PetscFunctionBegin;
  RDyFill(PetscInt, array, num_cells, value);
  PetscErrorCode(0);
}

PetscCall(RDyFillInt(cells->global_ids, num_cells, -1));
```

Comment (jeff-cohere) [reply swe_roe/ex2.c]: That's fine with me, if we'd like everything to have a return code. I was motivated by `PetscOptionsBegin` and `PetscOptionsEnd`, which are also macros that don't return error codes, so I don't think it's a big deal either way stylistically.

Comment (bishtgautam) [reply swe_roe/ex2.c]: In that case, let's have a return code. Everything else looks good.

File: swe_roe/ex2.c [hunk 2]
```diff
@@ -8,418 +8,461 @@ static char help[] = "Partial 2D dam break problem.\n";
...
-
-  RDyCoordinate *coordinate; /* (x,y,z) location of the vertex                       */
-} RDyVertex;
-
-typedef struct {
-  PetscInt *id;        /* id of the edge in local numbering         */
-  PetscInt *global_id; /* global id of the edge in local numbering */
-
-  PetscBool *is_local; /* true if the edge : (1) */
-                       /* 1. Is shared by locally owned cells, or   */
```
Comment (donghuix) [inline swe_roe/ex2.c:76]: Do we want to keep these comments? I think those are useful.

**Review by donghuix — COMMENTED** [review]
I only have one minor comment.

File: swe_roe/ex2.c [hunk 3]
```diff
@@ -8,418 +8,461 @@ static char help[] = "Partial 2D dam break problem.\n";
...
-  PetscCall(RDyAllocate_IntegerArray_1D(&edges->cell_ids, num_edges * num_cells));
+/// A struct of arrays storing information about edges separating mesh cells.
+/// The ith element in each array stores a property for edge i.
+typedef struct {
+  /// local IDs of edges in local numbering
+  PetscInt *ids;
+  /// global IDs of edges in local numbering
+  PetscInt *global_ids;
+
+  /// PETSC_TRUE iff edge is shared by locally owned cells, OR
```
Comment (jeff-cohere) [inline swe_roe/ex2.c:404]: @donghuix , here's the comment you were talking about. I didn't get rid of any comments--just marked them up with Doxygen tags and elaborated.

File: swe_roe/ex2.c (hunk 2 above)
Comment (jeff-cohere) [reply swe_roe/ex2.c:76]: All these comments are intact (though maybe re-expressed) elsewhere. I've pointed you to this one.

File: swe_roe/ex2.c (hunk 1 above)
Comment (jeff-cohere) [reply swe_roe/ex2.c]: I hacked a return code into the macro. This operation always succeeds unless you give it bad memory, which is very easy to troubleshoot.

**Review by bishtgautam — APPROVED** [review]

File: swe_roe/ex2.c (hunk 2 above)
Comment (donghuix) [reply swe_roe/ex2.c:76]: @jeff-cohere , thanks for the clarification.

**Review by donghuix — APPROVED** [review]

**Pull request merged**